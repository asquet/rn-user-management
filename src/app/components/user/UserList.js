import React from 'react';
import EntityList from '../common/list/EntityList/EntityList';
import UserListItem from './ListItem';

export default function UserList({ goToEdit, goToCreate, users, ...props }) {
  return (
    <EntityList
      data={users}
      onAdd={goToCreate}
      renderListItem={rowData => (
        <UserListItem
          {...rowData}
          onRowClick={() => {
            goToEdit(rowData.id);
          }}

        />
      )}
      {...props}
    />
  );
}

UserList.propTypes = {
  goToEdit: React.PropTypes.func.isRequired,
  goToCreate: React.PropTypes.func.isRequired,
  users: React.PropTypes.arrayOf(React.PropTypes.shape({})).isRequired,
};

