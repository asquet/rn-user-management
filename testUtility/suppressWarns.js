export default function suppressWarns() {
  // https://github.com/facebook/react-native/issues/12762
  console.error = jest.genMockFunction((error) => {
    if (error.match(/error/gi)) {
      throw new Error(error);
    }
  });
}
