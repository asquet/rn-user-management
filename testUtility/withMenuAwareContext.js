import withContext from './withContext';
import Menu from '../src/app/components/Menu';

export default function withMenuAwareContext(cmp) {
  const context = {
    menuOpen: () => {
    },
    menuClose: () => {
    },
  };
  return withContext({ context, contextType: Menu.childContextTypes })(cmp);
}
