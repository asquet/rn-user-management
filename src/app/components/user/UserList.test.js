import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import UserList from './UserList';
import withMenuAwareContext from '../../../../testUtility/withMenuAwareContext';
import suppressWarns from '../../../../testUtility/suppressWarns';

suppressWarns();

// Note: test renderer must be required after react-native.

describe('UserList', () => {
  const users = [
    { id: 1, name: 'user 1' },
    { id: 2, name: 'user 2' },
  ];
  it('renders correctly', () => {
    const Cmp = withMenuAwareContext(UserList);

    const tree = renderer.create(
      <Cmp
        onSearch={jest.fn()}
        data={users}
        searchString=""
        goToCreate={jest.fn()}
        goToEdit={jest.fn()}
      />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
