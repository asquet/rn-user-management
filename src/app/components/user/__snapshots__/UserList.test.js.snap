// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    closeOnRowBeginSwipe={false}
    closeOnRowPress={true}
    closeOnScroll={true}
    dataSource={
      ListViewDataSource {
        "items": 2,
      }
    }
    directionalDistanceChangeThreshold={2}
    disableLeftSwipe={false}
    disableRightSwipe={false}
    enableEmptySections={true}
    leftOpenValue={0}
    onScroll={[Function]}
    previewFirstRow={false}
    recalculateHiddenLayout={false}
    renderFooter={[Function]}
    renderHeader={[Function]}
    renderHiddenRow={[Function]}
    renderRow={[Function]}
    renderScrollComponent={[Function]}
    renderSeparator={[Function]}
    rightOpenValue={-80}
    swipeToOpenPercent={50}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "paddingRight": 8,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#393e42",
                "borderBottomColor": "#000",
                "borderBottomWidth": 1,
                "borderTopColor": "#000",
                "borderTopWidth": 1,
              },
              false,
              Object {
                "backgroundColor": "white",
                "borderBottomWidth": 0,
                "borderTopWidth": 0,
                "flex": 1,
                "height": 50,
                "margin": 0,
                "paddingBottom": 3,
                "paddingLeft": 3,
                "paddingRight": 3,
                "paddingTop": 3,
              },
            ]
          }
        >
          <TextInput
            autoCapitalize={undefined}
            autoCorrect={undefined}
            autoFocus={undefined}
            blurOnSubmit={undefined}
            clear={undefined}
            clearButtonMode={undefined}
            clearTextOnFocus={undefined}
            dataDetectorTypes={undefined}
            defaultValue={undefined}
            editable={undefined}
            enablesReturnKeyAutomatically={undefined}
            inlineImageLeft={undefined}
            inlineImagePadding={undefined}
            isFocused={undefined}
            keyboardAppearance={undefined}
            keyboardType={undefined}
            maxLength={undefined}
            multiline={undefined}
            numberOfLines={undefined}
            onBlur={undefined}
            onChange={undefined}
            onChangeText={[Function]}
            onContentSizeChange={undefined}
            onEndEditing={undefined}
            onFocus={undefined}
            onKeyPress={undefined}
            onLayout={undefined}
            onSelectionChange={undefined}
            onSubmitEditing={undefined}
            placeholder="Search..."
            placeholderTextColor="#86939e"
            returnKeyLabel={undefined}
            returnKeyType={undefined}
            secureTextEntry={undefined}
            selectTextOnFocus={undefined}
            selectionColor="#86939e"
            selectionState={undefined}
            style={
              Array [
                Object {
                  "backgroundColor": "#303337",
                  "borderRadius": 3,
                  "color": "#86939e",
                  "fontSize": 17.5,
                  "height": 30,
                  "margin": 8,
                  "overflow": "hidden",
                  "paddingLeft": 26,
                  "paddingRight": 19,
                },
                false,
                false,
                false,
                Object {
                  "backgroundColor": "#DDD",
                  "color": "#555",
                  "flex": 1,
                  "fontSize": 18,
                  "height": 44,
                  "margin": 0,
                  "paddingBottom": 10,
                  "paddingLeft": 30,
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#86939e",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "left": 16,
                    "position": "absolute",
                    "top": 15.5,
                  },
                  Object {
                    "fontSize": 24,
                    "left": 8,
                    "top": 13,
                  },
                ],
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "padding": 5,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "EvilIcons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 0,
                "width": 0,
              },
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#9E9E9E",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginLeft": 15,
                    "marginRight": 15,
                    "padding": 19,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "padding": 12,
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 3,
                      "padding": 5,
                    },
                    Object {
                      "alignSelf": "flex-end",
                      "backgroundColor": "red",
                      "borderRadius": 0,
                      "borderWidth": 0,
                      "flex": 0,
                      "height": 50,
                      "marginRight": 0,
                      "padding": 10,
                      "width": 80,
                    },
                  ],
                  undefined,
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": undefined,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 20,
                      },
                      undefined,
                      Object {
                        "fontSize": 17.5,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Delete
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Object {
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "flex": 1,
                  "flexDirection": "row",
                  "height": 50,
                  "paddingBottom": 5,
                  "paddingLeft": 5,
                  "paddingRight": 15,
                  "paddingTop": 5,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "borderRadius": 20,
                    "height": 40,
                    "marginLeft": 10,
                    "marginRight": 10,
                    "width": 40,
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#707070",
                      "fontSize": 18,
                    },
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Object {
                        "fontSize": 20,
                      },
                    ],
                  ]
                }
              >
                user 1
              </Text>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#9E9E9E",
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "marginLeft": 15,
                        "marginRight": 15,
                        "padding": 19,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Object {
                        "padding": 12,
                      },
                      Array [
                        Object {
                          "alignItems": "center",
                          "borderRadius": 3,
                          "padding": 5,
                        },
                        Object {
                          "backgroundColor": "white",
                          "borderColor": "#6699CC",
                          "borderWidth": 1,
                          "height": 40,
                          "paddingLeft": 5,
                          "paddingRight": 5,
                          "width": 40,
                        },
                      ],
                      undefined,
                      undefined,
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 18,
                        },
                        Array [
                          Object {
                            "marginRight": 10,
                          },
                          Object {
                            "color": "#707070",
                            "fontSize": 24,
                            "marginRight": 0,
                          },
                        ],
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontFamily": undefined,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Array [
                          Object {
                            "color": "white",
                            "fontSize": 20,
                          },
                          undefined,
                          Object {
                            "fontSize": 17.5,
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                        ],
                      ]
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 0,
                "width": 0,
              },
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#9E9E9E",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginLeft": 15,
                    "marginRight": 15,
                    "padding": 19,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "padding": 12,
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 3,
                      "padding": 5,
                    },
                    Object {
                      "alignSelf": "flex-end",
                      "backgroundColor": "red",
                      "borderRadius": 0,
                      "borderWidth": 0,
                      "flex": 0,
                      "height": 50,
                      "marginRight": 0,
                      "padding": 10,
                      "width": 80,
                    },
                  ],
                  undefined,
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": undefined,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 20,
                      },
                      undefined,
                      Object {
                        "fontSize": 17.5,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Delete
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Object {
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "flex": 1,
                  "flexDirection": "row",
                  "height": 50,
                  "paddingBottom": 5,
                  "paddingLeft": 5,
                  "paddingRight": 15,
                  "paddingTop": 5,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "borderRadius": 20,
                    "height": 40,
                    "marginLeft": 10,
                    "marginRight": 10,
                    "width": 40,
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#707070",
                      "fontSize": 18,
                    },
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Object {
                        "fontSize": 20,
                      },
                    ],
                  ]
                }
              >
                user 2
              </Text>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#9E9E9E",
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "marginLeft": 15,
                        "marginRight": 15,
                        "padding": 19,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Object {
                        "padding": 12,
                      },
                      Array [
                        Object {
                          "alignItems": "center",
                          "borderRadius": 3,
                          "padding": 5,
                        },
                        Object {
                          "backgroundColor": "white",
                          "borderColor": "#6699CC",
                          "borderWidth": 1,
                          "height": 40,
                          "paddingLeft": 5,
                          "paddingRight": 5,
                          "width": 40,
                        },
                      ],
                      undefined,
                      undefined,
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 18,
                        },
                        Array [
                          Object {
                            "marginRight": 10,
                          },
                          Object {
                            "color": "#707070",
                            "fontSize": 24,
                            "marginRight": 0,
                          },
                        ],
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontFamily": undefined,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Array [
                          Object {
                            "color": "white",
                            "fontSize": 20,
                          },
                          undefined,
                          Object {
                            "fontSize": 17.5,
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                        ],
                      ]
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 85,
          }
        }
      />
    </View>
  </RCTScrollView>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginLeft": 15,
            "marginRight": 15,
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "padding": 12,
          },
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 3,
              "padding": 5,
            },
            Object {
              "backgroundColor": "white",
              "borderRadius": 30,
              "bottom": 25,
              "height": 60,
              "justifyContent": "center",
              "position": "absolute",
              "right": 25,
              "width": 60,
            },
          ],
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": undefined,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              Object {
                "color": "blue",
              },
              Object {
                "fontSize": 17.5,
              },
              undefined,
              undefined,
              undefined,
              undefined,
            ],
          ]
        }
      >
        +
      </Text>
    </View>
  </View>
</View>
`;
